src/main/java/Edit_Group_Profile_Use_Case/EditGroupProfileDSGateway.java

```diff
@@ -0,0 +1,7 @@
+package Edit_Group_Profile_Use_Case;
+
+public interface EditGroupProfileDSGateway {
+    boolean existsByGroupName(String groupName);
+    void findGroup(String groupName);
+    void saveGroupProfile(EditGroupProfileDsRequestModel requestModel);
+}
```

Comment:
For better Clean Architecture, I think we could share a Group DS Gateway interface, as I also have existsbyIdentifier and save function. When we make changes to the Group Profile and therefore the Group entity, we have to re-save the entire hashmap of Groups in the system, so I believe all of your database related interfaces and functions should be the same as mine.